frame/vesting/src/lib.rs

```diff
@@ -195,7 +199,11 @@ pub mod pallet {
 
 		#[cfg(feature = "try-runtime")]
 		fn post_upgrade() -> Result<(), &'static str> {
-			migrations::v1::post_migrate::<T>()
+			if StorageVersion::<T>::get() == Releases::V1 {
+				migrations::v1::post_migrate::<T>()
+			} else {
+				Ok(())
+			}
```

Comment:
The migration should be successful and have upgrade the storage version
```suggestion
			assert_eq!(StorageVersion::<T>::get(), Releases::V1)
			migrations::v1::post_migrate::<T>()
```

Reply:
Ok i reverted that part all together. The StorageVersion is checked in post_migrate. I thought that v1::post_migrate should only get checked if we migrated from v0 -> v1 and not if there would be a potential v1 -> v2 migration. Didn't make a lot if sense since there is no v2 yet. 🤷🏻‍♂️

Reply:
yes when we have v2 we will need to upgrade the post_migrate